crates/networking/p2p/sync_manager.rs

```diff
@@ -105,6 +110,29 @@ impl SyncManager {
         self.syncer.try_lock().is_err()
     }
 
+    #[cfg(feature = "sync-test")]
+    async fn get_blocks_for_sync_test(store: &Store) -> Result<Option<H256>, StoreError> {
+        let get_latest = match env::var("SYNC-LATEST")
+            .expect("Failed to get sync configuration from environment")
+            .as_str()
+        {
+            "true" => true,
+            "false" => false,
+            _ => true,
+        };
+
+        let block_number = env::var("SYNC-BLOCK-NUM")
+            .expect("Failed to retrieve sync block number from environment")
+            .parse()
+            .expect("Error converting block number environmental variable to int");
+
+        if get_latest {
+            store.get_latest_canonical_block_hash().await
+        } else {
+            store.get_canonical_block_hash(block_number).await
+        }
+    }
+
```

Comment:
What do you think about moving this logic to `ethrex.rs`.  Then instead of changing where we start syncing (a fixed block instead of the latest canonical block) we would change the latest canonical block. This would have virtually the same behaviour (for full sync) while also being more isolated so that we don't risk breaking this when refactoring/adding logic to the `SyncManager`